### PR TITLE
Fixing bugs with Gene and metadata file arguments.

### PR DESCRIPTION
**Pull request by knapii-developments**
Bugs were introduced from PAPI with PR #20. Arguments for a specific file type are now passed individually. Arguments that are passed in as dictionary/json are interpreted as literals using ast.literals.

### CODE REVIEW

File: ingest/dense.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def __init__(self, file_path, file_id, study_accession, file_params):
         self.file_id = file_id
         self.study_accession = study_accession
         self.cell_names = self.get_next_line(increase_line_count=False)[1:]
-        self.file_params = file_params
+        self.matrix_params = {k: v for k, v in kwargs.items() if v is not None}
```
Comment (knapii-developments) [inline ingest/dense.py]: Remove key from dictionary that have value=None

Comment (eweitz) [inline ingest/dense.py]: Helpful -- I'd suggest adding that comment directly in the code.

```suggestion
        # Remove from dictionary any keys that have value=None
        self.matrix_params = {k: v for k, v in kwargs.items() if v is not None}
```

File: ingest/ingest_pipeline.py [hunk 2]
```diff
@@ -7,30 +7,29 @@
...
+python ingest_pipeline.py --study-accession SCP1 --file-id 123abc ingest_cluster --cluster-file ../tests/data/10k_cells_29k_genes.cluster.txt --ingest-cluster --name cluster1 --domain-ranges "{'x':[-1, 1], 'y':[-1, 1], 'z':[-1, 1]}"
 
 # Ingest Cell Metadata file
 python ingest_pipeline.py --study-accession SCP1 --file-id 123abc ingest_cell_metadata --cell-metadata-file ../tests/data/10k_cells_29k_genes.metadata.tsv --ingest-cell-metadata
 
 # Ingest dense file
-python ingest_pipeline.py  --study-accession SCP1 --file-id 123abc ingest_expression --file-params '{"taxon_name": "Homo sapiens", "taxon_common_name": "human", "ncbi_taxid": "9606", "genome_assembly_accession": "GCA_000001405.1", "genome_annotation": "Ensemble 94"}' --matrix-file ../tests/data/dense_matrix_19_genes_100k_cells.txt --matrix-file-type dense
+python ingest_pipeline.py  --study-accession SCP1 --file-id 123abc ingest_expression --taxon-name 'Homo Sapiens' --taxon-common-name humans --matrix-file ../tests/data/dense_matrix_19_genes_100k_cells.txt --matrix-file-type dense
```
Comment (eweitz) [inline ingest/ingest_pipeline.py]: Copyediting example for capitalization ("Homo Sapiens" -> "Homo sapiens") and grammatical number ("humans" -> "human").  Also, wherever `--taxon-name` and `--taxon-common-name` are used, `--ncbi-taxid` is also required.

```suggestion
python ingest_pipeline.py  --study-accession SCP1 --file-id 123abc ingest_expression --taxon-name 'Homo sapiens' --taxon-common-name human --ncbi-taxid 9606 --matrix-file ../tests/data/dense_matrix_19_genes_100k_cells.txt --matrix-file-type dense
```

And similarly for MTX example.

File: ingest/ingest_pipeline.py [hunk 3]
```diff
@@ -308,9 +285,22 @@ def create_parser():
...
-        help="Optional parameters for expression files",
+        "--taxon-name",
+        help="Taxon (scientific) name associated with file. Ex. 'Homo Sapiens' ",
+    )
+    parser_ingest_expression.add_argument(
+        "--taxon-common-name",
+        help="Taxon common name associated with study. Ex. 'humans'",
+    )
+    parser_ingest_expression.add_argument(
+        "--ncbi-taxid", help="NCBI ID associated with file"
```
Comment (eweitz) [inline ingest/ingest_pipeline.py]:
```suggestion
        "--ncbi-taxid", help="NCBI Taxonomy ID of taxon associated with file.  E.g. '9606'"
```

File: ingest/ingest_pipeline.py [hunk 4]
```diff
@@ -308,9 +285,22 @@ def create_parser():
...
     parser_ingest_expression.add_argument(
-        "--file-params",
-        type=json.loads,
-        help="Optional parameters for expression files",
+        "--taxon-name",
+        help="Taxon (scientific) name associated with file. Ex. 'Homo Sapiens' ",
+    )
+    parser_ingest_expression.add_argument(
+        "--taxon-common-name",
+        help="Taxon common name associated with study. Ex. 'humans'",
```
Comment (eweitz) [inline ingest/ingest_pipeline.py]:
```suggestion
        help="Common name of taxon associated with file.  E.g. 'human'",
```

File: ingest/ingest_pipeline.py [hunk 5]
```diff
@@ -308,9 +285,22 @@ def create_parser():
         <gene file path>. See --help for more information"
 
     parser_ingest_expression.add_argument(
-        "--file-params",
-        type=json.loads,
-        help="Optional parameters for expression files",
+        "--taxon-name",
+        help="Taxon (scientific) name associated with file. Ex. 'Homo Sapiens' ",
```
Comment (eweitz) [inline ingest/ingest_pipeline.py]:
```suggestion
        help="Scientific name of taxon associated with file.  E.g. 'Homo sapiens'",
```

File: ingest/ingest_pipeline.py [hunk 6]
```diff
@@ -308,9 +285,22 @@ def create_parser():
...
+    )
+    parser_ingest_expression.add_argument(
+        "--ncbi-taxid", help="NCBI ID associated with file"
+    )
+    parser_ingest_expression.add_argument(
+        "--genome-assembly-accession", help="Genome assembly accession for file."
+    )
+    parser_ingest_expression.add_argument(
+        "--genome-annotation",
+        help="Genomic annotation for expression files. Ex. 'Ensemble 94'",
```
Comment (eweitz) [inline ingest/ingest_pipeline.py]:
```suggestion
        help="Genomic annotation for expression files.  E.g. 'Ensembl 94'",
```

File: ingest/ingest_pipeline.py [hunk 7]
```diff
@@ -308,9 +285,22 @@ def create_parser():
...
+    )
+    parser_ingest_expression.add_argument(
+        "--taxon-common-name",
+        help="Taxon common name associated with study. Ex. 'humans'",
+    )
+    parser_ingest_expression.add_argument(
+        "--ncbi-taxid", help="NCBI ID associated with file"
+    )
+    parser_ingest_expression.add_argument(
+        "--genome-assembly-accession", help="Genome assembly accession for file."
```
Comment (eweitz) [inline ingest/ingest_pipeline.py]: Examples are given in related arguments' `help` field (which is excellent), so for consistency:

```suggestion
        "--genome-assembly-accession", help="Genome assembly accession for file.  E.g. 'GCA_000001405.15'"
```

File: tests/test_ingest.py [hunk 8]
```diff
@@ -120,8 +120,16 @@ def test_ingest_dense_matrix(self):
...
+            "--taxon-name",
+            "Homo sapiens",
+            "--taxon-common-name",
+            "human",
+            "--ncbi-taxid",
+            "9606",
+            "--genome-assembly-accession",
+            "GCA_000001405.15",
+            "--genome-annotation",
+            "Ensemble 94",
```
Comment (eweitz) [inline tests/test_ingest.py]: I'd suggest a global find-replace through all files to change "Ensemble 94" to "Ensembl 94".

```suggestion
            "Ensembl 94",
```

**Pull request merged**